packages/transactions/src/lifetime.ts

```diff
@@ -102,3 +106,63 @@ export type SetTransactionLifetimeFromTransactionMessage<
           ? TransactionWithDurableNonceLifetime & TTransaction
           : TransactionWithLifetime & TTransaction
     : TTransaction;
+
+const SYSTEM_PROGRAM_ADDRESS = '11111111111111111111111111111111' as Address;
+
+function compiledInstructionIsAdvanceNonceInstruction(
+    instruction: CompiledTransactionMessage['instructions'][number],
+    staticAddresses: Address[],
+): instruction is typeof instruction & { accountIndices: [number, number, number] } {
+    return (
+        staticAddresses[instruction.programAddressIndex] === SYSTEM_PROGRAM_ADDRESS &&
+        // Test for `AdvanceNonceAccount` instruction data
+        instruction.data != null &&
+        isAdvanceNonceAccountInstructionData(instruction.data) &&
+        // Test for exactly 3 accounts
+        instruction.accountIndices?.length === 3
+    );
+}
+
+function isAdvanceNonceAccountInstructionData(data: ReadonlyUint8Array): boolean {
+    // AdvanceNonceAccount is the fifth instruction in the System Program (index 4)
+    return data.byteLength === 4 && data[0] === 4 && data[1] === 0 && data[2] === 0 && data[3] === 0;
+}
+
+/**
+ * Check if a compiled instruction is an AdvanceNonceAccount instruction
+ * @param compiledTransactionMessage A compiled transaction message that includes a lifetime token
+ * @returns True if the instruction is an AdvanceNonceAccount instruction, false otherwise
+ * Note that this is less precise than checking a decompiled instruction, as we can't inspect
+ * the address or role of input accounts (which may be in lookup tables). However, this is
+ * sufficient for all valid advance durable nonce instructions.
+ * Note that the program address must not be in a lookup table, see [this answer on StackExchange](https://solana.stackexchange.com/a/16224/289)
+ * @see {@link isAdvanceNonceAccountInstruction}
+ * Note that this function is async to allow for future implementations that may fetch `lastValidBlockHeight` using an RPC
+ */
+// eslint-disable-next-line @typescript-eslint/require-await
+export async function getTransactionLifetimeConstraintFromCompiledTransactionMessage(
```

Comment:
Sorry if I've missed something but it doesn't seem like this function needs to be asynchronous. Is it to future proof it?

Reply:
Gotcha, thank you for clarifying. 🙏